ember-app/app/components/hb-milestone-card.js

```diff
@@ -34,11 +21,42 @@ var HbMilestoneCard = HbCard.extend({
       });
     });
 
-    takeWhile(columns, (x) => {return !get(x, 'selected');}, this)
-      .forEach((x) => { set(x, 'selected', true); });
+    var selected = columns.findBy('selected');
+    var powerbar_total = columns.indexOf(selected);
+    var visible_columns = columns.slice(0, this.powerBarLength);
+
+    // Calculate the powerbar progress proportionally
+    visible_columns = this.proportionalProgress(visible_columns, powerbar_total, columns.length);
 
-    return columns;
-  }.property('issue.data.current_state', 'taskColumns')
+    return visible_columns;
+  }.property('issue.data.current_state', 'taskColumns'),
+  proportionalProgress: function(columns, powerbar_total, total_length){
+    // If powerbar is on first column, as meters are empty
+    if(!powerbar_total){ columns.setEach('selected', false); return columns; }
+
+    // If powerbar is on last column, all meters are full
+    if(powerbar_total === (total_length - 1)){
+      columns.setEach('selected', true);
+      return columns;
+    }
+
+    // Otherwise calculate the proportional progress of the meter
+    var percent_complete = (powerbar_total - 1) / total_length;
+    var selected_columns_index = Math.ceil(percent_complete * this.powerBarLength);
```

Comment:
Since you are already handling the full meter scenario, I think the math works out if you change this to `Math.ceil(percent_complete * (this.powerBarLength - 1))` (essentially splitting between all but the last column).

The `powerbar_total - 1` on the line above surprises me. Isn't `powerbar_total` already a 0-based index?